common/src/java/com/zimbra/common/httpclient/HttpClientUtil.java

```diff
@@ -67,16 +68,40 @@ public static HttpResponse executeMethod(HttpClient client, HttpRequestBase meth
                 .setDefaultCookieStore(cookieStore)
                 .build ();
             return httpClient.execute(method);
-           
+
         }
         return client.execute(method);
     }
-    
+
+    public static HttpResponse executeMethod(HttpClientBuilder clientBuilder, HttpRequestBase method, BasicCookieStore state, HttpClientContext context) throws HttpException, IOException {
+        ProxyHostConfiguration proxyConfig = HttpProxyConfig.getProxyConfig(method.getURI().toString());
+        BasicCookieStore cookieStore = new  BasicCookieStore();
+        if (state != null) {
+            cookieStore = state;
+            clientBuilder.setDefaultCookieStore(cookieStore);
+        }
+        CloseableHttpClient httpClient = clientBuilder.build();
```

Comment:
Just checking, we don't need `httpClient.close()` for this method because the connection manger is shared or something?

Reply:
Yes, the clientbuilder and default configuration is shared